components/producers/github-codeql/main.go

```diff
@@ -49,27 +63,36 @@ func main() {
 	}
 }
 
-func listAlertsForRepo(owner, repo, token string) ([]*github.Alert, error) {
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: token},
-	)
-	oAuthClient := oauth2.NewClient(context.Background(), ts)
-	apiClient := github.NewClient(oAuthClient)
-
+func listAlertsForRepo(apiClient wrapper.Wrapper, owner, repo, toolName string) ([]*github.Alert, error) {
+	var severity string
+	if Severity != "" {
+		severity = Severity
+	}
 	opt := &github.AlertListOptions{
-		State: "open",
+		State:    "open",
+		Ref:      Ref,
+		Severity: severity,
+		ToolName: toolName,
 		ListOptions: github.ListOptions{
 			PerPage: 30,
 		},
 	}
 
 	var allAlerts []*github.Alert
 	for {
-		alerts, resp, err := apiClient.CodeScanning.ListAlertsForRepo(context.Background(), owner, repo, opt)
+		alerts, resp, err := apiClient.ListRepoAlerts(context.Background(), owner, repo, opt)
 		if err != nil {
 			return nil, err
 		}
 
+		if resp.StatusCode > 299 {
+			body, bodyReadErr := io.ReadAll(resp.Body)
+			if bodyReadErr != nil {
+				return nil, errors.Errorf("could not read response error from github, err:%w", bodyReadErr)
+			}
+			return nil, errors.Errorf("did not receive a valid status code from github, status code: %d, body:%s", resp.StatusCode, body)
```

Comment:
usually, it's just some json with a short but helpful error message, i prefer it in the error since the error is our logs but i can also add a log line